# Descriptive Statistics Assignment

In [95]:
import pandas as pd

data_url = "http://data.insideairbnb.com/united-states/ny/new-york-city/2020-06-08/visualisations/listings.csv"

### Read the aggregated AirBnB listings of New York (using `data_url`) into a Pandas data frame.

In [96]:
airbnb = pd.read_csv(data_url)
airbnb.head()

id  ... availability_365
0  2060  ...              365
1  2595  ...              335
2  3831  ...              276
3  5099  ...                0
4  5121  ...              365

[5 rows x 16 columns]

### Print the data type for each column.

In [97]:
airbnb.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                               int64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
dtype: object

### Generate a basic set of descriptive statistics for the numeric fields in the data set.

In [98]:
airbnb_stats = airbnb.describe().T
airbnb_stats

count  ...           max
id                              49530.0  ...  4.370336e+07
host_id                         49530.0  ...  3.490826e+08
latitude                        49530.0  ...  4.091169e+01
longitude                       49530.0  ... -7.371299e+01
price                           49530.0  ...  1.000000e+04
minimum_nights                  49530.0  ...  1.250000e+03
number_of_reviews               49530.0  ...  7.460000e+02
reviews_per_month               38211.0  ...  5.380000e+01
calculated_host_listings_count  49530.0  ...  2.800000e+02
availability_365                49530.0  ...  3.650000e+02

[10 rows x 8 columns]

### Calculate the following for each numeric field in the data set and add them to the data frame containing the basic set of descriptive statistics.

- Variance
- Mean Absolute Deviation
- Range
- Interquartile Range

In [106]:
# create column for variance
airbnb_stats['var'] = airbnb.var()

# create column for mean absolute deviation
airbnb_stats['mad'] = airbnb.mad()

# create column for range
airbnb_stats['range'] = airbnb_stats['max'] - airbnb_stats['min']

# create column for interquartile range
airbnb_stats['IQR'] = airbnb_stats['75%'] - airbnb_stats['25%']

airbnb_stats.head()

count          mean  ...         range           IQR
id         49530.0  2.295964e+07  ...  4.370130e+07  2.472729e+07
host_id    49530.0  8.509951e+07  ...  3.490804e+08  1.280898e+08
latitude   49530.0  4.072924e+01  ...  4.119000e-01  7.297000e-02
longitude  49530.0 -7.395104e+01  ...  5.278500e-01  4.907750e-02
price      49530.0  1.626439e+02  ...  1.000000e+04  1.070000e+02

[5 rows x 12 columns]

### Generate a basic set of descriptive statistics for the categorical fields in the data set.

In [74]:
# generate and transpose descriptive stats for categorical fields
df.describe(include='object').T

count unique                  top   freq
name                 49512  48247  Home away from home     21
host_name            49524  11572              Michael    416
neighbourhood_group  49530      5            Manhattan  21963
neighbourhood        49530    222         Williamsburg   3752
room_type            49530      4      Entire home/apt  25716
last_review          38211   2077           2020-01-01    940

### Calculate the Pearson correlation coefficients between price, minimum_nights, reviews_per_month, calculated_host_listings_count and availability_365

In [42]:
df[['price', 'minimum_nights', 'reviews_per_month', 'calculated_host_listings_count', 'availability_365']].corr()

price  ...  availability_365
price                           1.000000  ...          0.055788
minimum_nights                  0.015923  ...          0.148435
reviews_per_month              -0.011623  ...          0.213093
calculated_host_listings_count  0.004730  ...          0.173181
availability_365                0.055788  ...          1.000000

[5 rows x 5 columns]

### Group the data by neighbourhood and calculate the mean and standard deviation of price for each neighbourhood.

In [122]:
neighbourhoods = airbnb.groupby(['neighbourhood']).agg({'price': ['mean', 'std']})
neighbourhoods


price             
                        mean          std
neighbourhood                            
Allerton           94.500000    83.219379
Arden Heights     102.600000    54.834296
Arrochar          104.315789   136.241229
Arverne           187.125000   231.710658
Astoria           104.109865   124.606979
...                      ...          ...
Windsor Terrace   138.067114    89.830596
Woodhaven          69.638095    43.294191
Woodlawn           61.700000    24.882167
Woodrow          2350.000000  2333.452378
Woodside           69.383721    85.328235

[222 rows x 2 columns]

### Create a new mimimum revenue index variable that combines the price, minimum nights, calculate host listings count into a single value for each dwelling. Then sort the data frame by that value to see which dwelling has the most financial strength. 



In [119]:
airbnb['MRI'] = (airbnb['price'] * airbnb['minimum_nights'] * airbnb['calculated_host_listings_count'])

sort_by_mri = airbnb.sort_values(by='MRI', ascending=False).reset_index()

sort_by_mri.head()

index        id  ... min_rev_index       MRI
0  14754  13443640  ...      10956000  10956000
1  31916  30389115  ...       8467200   8467200
2   9961   8810949  ...       7768800   7768800
3  33929  32399523  ...       6804000   6804000
4  31992  30394518  ...       5275200   5275200

[5 rows x 19 columns]

### Group the data by neighbourhood_group and calculate the average minimum revenue index for each neighbourhood_group.

In [124]:
neighbourhood_groups = airbnb.groupby(['neighbourhood_group']).agg({'MRI': ['mean']})
neighbourhood_groups

MRI
                             mean
neighbourhood_group              
Bronx                  987.498331
Brooklyn              3434.846270
Manhattan            48328.517097
Queens                7537.764338
Staten Island         1201.845946

# Identify 3 scenarios in your everyday life where you use probability.
  1. Going to the gym: will I get a machine? 
  2. Will I use my umbrella in AZ?
  3. Will my partner like the gift I get them?

# Using the formula you learned to calculate probabilities, approximate the probabilities of the events in each of those 3 scenarios.
### P(E) = Event Frequency / Total Observations
  1. P(getting a machine) = how often I've gotten a machine / total number of times I've tried
  
    P(getting a machine) = 80%  

  2. P(I'll use my umbrella) = how often it's rained in AZ / number of days I've observed
  
    P(I'll use my umbrella) = 3%

  3. P(my partner likes the gift) = how many times they've liked my gifts / how many gifts I've given
  
    P(my partner likes the gift) = 60%

#Think about what potential dependencies the events in each of your 3 scenarios have. Calculate the joint probabilities for the events and their dependencies.
###P(AB) = P(A|B) x P(B)

  1. Whether or not the gym is open is a dependency:
  
    A|B: Getting a machine given then gym is open (80%) B: gym is open (90%)
    
    P(the gym is open and you get a machine) = .8 * .9 = .72
    
    72% chance the gym will be open and you'll get a machine


  2. If I can fit an umbrella in my bag is a dependency:

    A|B: I'll need an umbrella given I can fit it in my bag (3%) B: I can fit it in my bag (50%)

    P(I'll need an umbrella and it will fit) = .03 * .5 = .015

    1.5% chance that I'll need an umbrella and will be able to fit it in my bag

  3. A|B: partner likes gift, given it arrives on time (60%) B: it arrives on time (20%)

    P(partner likes gift and it arrives on time) = .6 * .2 = .12

    12% chance that the gift arrives on time and my partner likes it

#Calculate the conditional probabilities for the events given their dependencies.
###P(A|B) = P(AB) / P(B)

1. AB: Getting a machine & gym is open (72%) B: Gym is open (90%)

    P(getting a machine given that the gym is open) = .72 / .9 = .8

2. AB: I use my umbrella & it fits in my bag (1.5%) B: it fits in my bag (50%)
    
    P(I use my umbrella given that it fits in my bag) = .015 / .5 = .03

3. AB: that the gift arrives on time and my partner likes it (12%) B: it arrives on time (20%)

    P(my partner likes the gift, given that it arrives on time) = .12 / .2 = .6
    
